### PR TITLE
[#174189493] Review CosmosdbModelVersioned public interface

### PR DESCRIPTION
**Pull request by balanza**
This introduces new methods and breaking changes to old ones for the purpose of handling different scenarios explicitly when saving versioned documents.

### Changes
* ~type `NewVersionedModel` doesn't include `version` field anymore.~
* type `NewVersionedModel` has been removed. `TN` is actually `T`, but it can be extended (to add tags for example).
* `create(o:TN)` creates a new document on db which version is always `0`.
* `upsert(o:TN)` creates a new document on db which version is calculated from the latest document present on db for the same modeld (`0` if no documents are found)
* `update(o: TR)` creates a new document on db which version is calculated as `o.version + 1`

### CODE REVIEW

File: src/utils/cosmosdb_model_versioned.ts [hunk 1]
```diff
@@ -223,4 +207,35 @@ export abstract class CosmosdbModelVersioned<
...
+    );
+  }
+
+  /**
+   * Strips off meta fields which are nor part of the base model definition
+   * @param o
+   */
+  private toBaseType(o: TR): T {
+    const { _etag, _rid, _self, _ts, id, version, ...n } = o;
+    return (n as unknown) as T;
```
Comment (BurnedMarshal) [inline src/utils/cosmosdb_model_versioned.ts:239]: Can have Type `T` an `id` field?

Comment (balanza) [reply src/utils/cosmosdb_model_versioned.ts:239]: why?

Comment (gunzip) [inline src/utils/cosmosdb_model_versioned.ts:239]: cn we use decode instead of casting?

Comment (BurnedMarshal) [reply src/utils/cosmosdb_model_versioned.ts:239]: This method removes the `id` field from the TR. But maybe `id` is reserved in a versioned model.

File: src/utils/__tests__/cosmosdb_model_versioned.test.ts [hunk 2]
```diff
@@ -124,135 +128,130 @@ const errorResponse: ErrorResponse = new Error();
...
-    const expectedNextVersion = 3;
-
+    const expectedNextVersion = 0;
+    const modelId = aNewMyDocument[aModelIdField];
     const model = new MyModel(container);
 
-    const result = await model
-      .upsert({ ...aNewMyDocument, version: modelCurrentVersion })
-      .run();
+    const result = await model.create(aNewMyDocument).run();
```
Comment (gunzip) [inline src/utils/__tests__/cosmosdb_model_versioned.test.ts:207]: this needs to be changed

File: src/utils/cosmosdb_model_versioned.ts (hunk 1 above)
Comment (balanza) [reply src/utils/cosmosdb_model_versioned.ts:255]: In versioned models `id` is a meta-data field, as it doesn't ship data and it's basically a Cosmos thing. I don't like the manual cast but I see no other way to skim unwanted fields.

**Pull request merged**